app/src/main/kotlin/org/eclipse/kuksa/testapp/KuksaDataBrokerActivity.kt

```diff
@@ -282,4 +285,35 @@ class KuksaDataBrokerActivity : ComponentActivity() {
             },
         )
     }
+
+    private fun loadVssPathSuggestions() {
+        val property = Property("Vehicle", listOf(Field.FIELD_VALUE))
+
+        dataBrokerEngine.fetch(
+            property,
+            object : CoroutineCallback<GetResponse>() {
+                override fun onSuccess(result: GetResponse?) {
+                    val entriesList = result?.entriesList
+                    val paths = entriesList?.map { it.path } ?: emptyList()
+
+                    val pathSet = TreeSet<String>()
+                    paths.forEach {
+                        pathSet.add(it)
+
+                        var value = it
+                        while (value.indexOf(".") > -1) {
```

Comment:
Yes, that looks fine. Could be OK for now to keep the createVssPathHierarchy inside the Activity class but maybe we could write an extension for the GetResponse or List<Types.DataEntry> to convert it into a VssPathHierarchy TreeSet? The activity class is already growing :)

Reply:
Maybe this is more like a "generateSuggestions(List<String>)" method. Then we could put it into the ViewModel?We could even think about creating a VssPath model to make it more clearer where the suggestions are generated from because the algorithm is explicitly checking for "." to split (optional or something for the future).